### PR TITLE
[cmake] Set macOS/iOS app bundle properties

### PR DESCRIPTION
**Pull request by antiagainst**
These properties are needed to generate `Info.plist` associated with macOS/iOS apps.

Fixes https://github.com/iree-org/iree/issues/11733

### CODE REVIEW

File: build_tools/cmake/iree_cc_binary.cmake [hunk 1]
```diff
@@ -192,4 +192,16 @@ function(iree_cc_binary)
...
+  # Set up Info.plist properties when building macOS/iOS app bundles.
+  get_target_property(APPLE_BUNDLE ${_NAME} MACOSX_BUNDLE)
+  if (APPLE_BUNDLE)
+    set_target_properties(${_NAME} PROPERTIES
+      MACOSX_BUNDLE_GUI_IDENTIFIER "com.google.iree.${_RULE_NAME}"
+      # These are just placeholder version numbers until we define proper
+      # version scheme and support.
+      MACOSX_BUNDLE_BUNDLE_VERSION 0.1
+      MACOSX_BUNDLE_SHORT_VERSION_STRING 0.1
+      MACOSX_BUNDLE_LONG_VERSION_STRING 0.1)
```
Comment (ScottTodd) [inline build_tools/cmake/iree_cc_binary.cmake:205]: Should we also set `MACOSX_BUNDLE_BUNDLE_NAME` and `MACOSX_BUNDLE_COPYRIGHT`? https://cmake.org/cmake/help/latest/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.html

```suggestion
    set_target_properties(${_NAME} PROPERTIES
      MACOSX_BUNDLE_BUNDLE_NAME "${_RULE_NAME}"
      MACOSX_BUNDLE_GUI_IDENTIFIER "com.google.iree.${_RULE_NAME}"
      MACOSX_BUNDLE_COPYRIGHT "Copyright © 2023 The IREE Authors"
      # These are just placeholder version numbers until we define proper
      # version scheme and support.
      MACOSX_BUNDLE_BUNDLE_VERSION 0.1
      MACOSX_BUNDLE_SHORT_VERSION_STRING 0.1
      MACOSX_BUNDLE_LONG_VERSION_STRING 0.1)
```

Comment (antiagainst) [reply build_tools/cmake/iree_cc_binary.cmake]: Yes! Thanks. :D

**Review by ScottTodd — APPROVED** [review]

File: build_tools/cmake/iree_cc_binary.cmake [hunk 2]
```diff
@@ -192,4 +192,18 @@ function(iree_cc_binary)
       )
     endif()
   endif()
+
+  # Set up Info.plist properties when building macOS/iOS app bundles.
+  get_target_property(APPLE_BUNDLE ${_NAME} MACOSX_BUNDLE)
+  if (APPLE_BUNDLE)
+    set_target_properties(${_NAME} PROPERTIES
+      MACOSX_BUNDLE_BUNDLE_NAME "${_RULE_NAME}"
+      MACOSX_BUNDLE_GUI_IDENTIFIER "com.google.iree.${_RULE_NAME}"
```
Comment (wangkuiyi) [inline build_tools/cmake/iree_cc_binary.cmake:201]: Sorry for the late reminder -- do we want to use `org.iree.${_RULE_NAME}` here instead?

Comment (antiagainst) [reply build_tools/cmake/iree_cc_binary.cmake:201]: I'm fine with it. Feel free to send a pull request! :)

Comment (stellaraccident) [reply build_tools/cmake/iree_cc_binary.cmake:201]: We actually own the iree.dev domain and should probably use that.

**Pull request merged**